examples/ui-demo/src/components/shared/MintCard.tsx

```diff
@@ -0,0 +1,241 @@
+"use client";
+import Image from "next/image";
+import { CheckIcon } from "../icons/check";
+import { GasIcon } from "../icons/gas";
+import { DrawIcon } from "../icons/draw";
+import { ReceiptIcon } from "../icons/receipt";
+import React, { useCallback, useState } from "react";
+import { hexToRGBA } from "../../utils/hexToRGBA";
+import { LoadingIcon } from "../icons/loading";
+import { ExternalLinkIcon } from "../icons/external-link";
+import {
+  useSendUserOperation,
+  useSmartAccountClient,
+} from "@account-kit/react";
+import { AccountKitNftMinterABI, nftContractAddress } from "@/utils/config";
+import { encodeFunctionData } from "viem";
+
+type NFTLoadingState = "loading" | "success";
+
+const initialState = {
+  signing: "signing",
+  gas: "gas",
+  batch: "batch",
+} satisfies mintStatus;
+
+type mintStatus = {
+  signing: NFTLoadingState | "signing";
+  gas: NFTLoadingState | "gas";
+  batch: NFTLoadingState | "batch";
+};
+
+export const MintCard = () => {
+  const [status, setStatus] = useState<mintStatus>(initialState);
+  // To be wired into the toast pr
+  const [hasError, setHasError] = useState(false);
+  const [hasCollected, setHasCollected] = useState(false);
+  const handleSuccess = () => {
+    setStatus((prev) => ({ ...prev, batch: "success" }));
```

Comment:
if we move the loading states into the `onMutate` then this should probably set all of the states to success in case of race conditions

Reply:
With the current implementation the request does not being until the other stages are complete.  We can adjust this if we decide to make them all resolve at once.